pkg/controller/service/service_controller.go

```diff
@@ -559,20 +559,21 @@ func getPortsForLB(service *v1.Service) ([]*v1.ServicePort, error) {
 		if protocol == "" {
 			protocol = sp.Protocol
 		} else if protocol != sp.Protocol && wantsLoadBalancer(service) {
-			// TODO:  Convert error messages to use event recorder
-			return nil, fmt.Errorf("mixed protocol external load balancers are not supported")
+			s.eventRecorder.Eventf(service, v1.EventTypeWarning, "mixed protocol external load balancers are not supported",
```

Comment:
This is a strange place to do validation like this (this function is only triggered when comparing an updated service object). Also, on the above it says the cloud provider itself is responsible for the protocol validation. Why don't we remove the validation here and delegate `EnsureLoadBalancer()` for doing the check of mixed protocol?

Reply:
Yeah~, mixed protocol may be checked in the function EnsureLoadBalancer().